components/event-receiver/org.wso2.carbon.event.receiver.core/src/main/java/org/wso2/carbon/event/receiver/core/internal/type/text/TextInputMapper.java

```diff
@@ -243,18 +243,21 @@ private int getEventAttributeLocation(InputMappingAttribute inputMappingAttribut
         } else if (EventReceiverUtil.isCorrelationAttribute(inputMappingAttribute.getToElementKey())) {
             for (Attribute correlationData : streamDefinition.getCorrelationData()) {
                 if (EventReceiverConstants.CORRELATION_DATA_PREFIX.concat(correlationData.getName()).equals(inputMappingAttribute.getToElementKey())) {
-                    return attributeLocation + streamDefinition.getMetaData().size();
+                    if (streamDefinition.getMetaData() != null) {
+                        attributeLocation += streamDefinition.getMetaData().size();
+                    }
+                    return attributeLocation;
```

Comment:
@grainier Can we get the meta data size calculation out of the for loop so that the if condition and retrieval of meta data size does not repeat for each iteration? i.e. outside of loop, do something like `metaSize = streamDefinition.getMetaData() == null ? 0: streamDefinition.getMetaData().size()`, then within the loop, simply do a `return attributeLocation + metaSize;`. However, since this code does not seem to be in the critical path, let's do it only if doing such an optimization does not compromise readability. WDYT?